lib/ABI/FunctionType.cpp

```diff
@@ -613,12 +647,60 @@ convertToCABIFunctionType(const model::RawFunctionType &Function,
 }
 
 std::optional<model::RawFunctionType>
-convertToRawFunctionType(const model::CABIFunctionType &Function,
-                         model::Binary &TheBinary) {
+tryConvertToRaw(const model::CABIFunctionType &Function,
+                model::Binary &TheBinary) {
   revng_assert(Function.ABI != model::ABI::Invalid);
   return skippingEnumSwitch<1>(Function.ABI, [&]<model::ABI::Values A>() {
     return ConvertionHelper<A>::toRaw(Function, TheBinary);
   });
 }
 
-} // namespace model
+Layout layout(const model::CABIFunctionType &Function) {
+  auto R = skippingEnumSwitch<1>(Function.ABI, [&]<model::ABI::Values A>() {
+    return ConvertionHelper<A>::getLayout(Function);
+  });
+  return Layout(std::move(R.Arguments),
+                std::move(R.ReturnValue),
+                std::move(R.CalleeSavedRegisters),
+                std::move(R.FinalStackOffset));
+}
+
+Layout layout(const model::RawFunctionType &Function) {
+  // Lay register arguments out.
+  std::decay<decltype(Layout::Arguments)>::type ResultArguments;
+  for (const model::NamedTypedRegister &Register : Function.Arguments)
+    ResultArguments.emplace_back().Registers = { Register.Location };
+
+  // Lay stack arguments out.
+  const model::Type *OriginalStackType = Function.StackArgumentsType.get();
+  auto *StackStruct = llvm::dyn_cast<model::StructType>(OriginalStackType);
+  revng_assert(StackStruct,
+               "`RawFunctionType::StackArgumentsType` must be a struct.");
+  for (const model::StructField &Field : StackStruct->Fields) {
+    typename Layout::Argument::StackSpan StackSpan;
+    StackSpan.Offset = Field.Offset;
+
+    auto MaybeSize = Field.Type.size();
+    revng_assert(MaybeSize.has_value());
+    StackSpan.Size = MaybeSize.value();
+
+    ResultArguments.emplace_back().Stack = std::move(StackSpan);
+  }
```

Comment:
Can we have a single argument for all of these?

Reply:
What do you mean? This creates `StackStruct->Fields.size()` new arguments.

Reply:
Done: https://github.com/revng/revng/pull/131/commits/7d430547c41935a010d2ff9cc708475f1f2590d8.